chain/chain/src/chain.rs

```diff
@@ -3709,6 +3712,9 @@ impl<'a> ChainUpdate<'a> {
                 receipt_proof_response.1,
             );
         }
+        if !apply_result.receipts_to_restore.is_empty() {
+            self.chain_store_update.save_receipts(&apply_result.receipts_to_restore);
+        }
```

Comment:
I don't think we need this here. If we state sync past that specific block, we do not have the block in the first place and therefore it does not make sense for us to restore receipts here.